### PR TITLE
Match Report Header and LHN Avatars 

### PR DESCRIPTION
**Pull request by grgia**
### Details

As part of https://github.com/Expensify/App/issues/20486, we want to ensure that the avatars match between report headers and LHN. This PR uses the logic from the LHN Options List to display the header avatars.

### Fixed Issues

$ https://github.com/Expensify/App/issues/22468
https://github.com/Expensify/App/issues/20486
PROPOSAL: 

### Tests

- [ ] Verify that no errors appear in the JS console

### QA Steps

- [ ] Verify that no errors appear in the JS console

Step 1. Create one of each of the following reports:
- [ ] IOU Report
- [ ] IOU Request
- [ ] Workspace Expense Report
- [ ] Workspace Expense Request

Step 2.
- [ ] Open each report and ensure that the avatar style in the header matches the style in the LHN

### IOUs
1. Start a DM between userA and userB.
2. Click on the plus actions icon next to "write something.." in the compose message box and select "request money".
3. Click on the expense request preview. 
4. In the expense report, confirm the subheader says `From <userA full name>`
image

### IOUs children  (IOU Requests)
1. Continuing from previous. Click on the specific expense in the expense report.
2. Confirm you see the single expense request view. Confirm the subheader says `From <parent IOU report name>` 
The parent IOU report can be seen in the LHN. In the below screenshot the parent IOU report name is "Marc Glasser owes $20.00" and so the child has that as its subheader.
image
3. Click on the subheader and confirm you are navigated up one level to parent. 
4. Create another thread in the expense report, Or a few nested thread. Confirm all thread subheaders show `From <parent IOU report name>`.
image

### Expense Reports
1. Create a workspace. As admin userA, invite userB to the workspace.
2. You should see a workspace chat between the two like so (view for the admin is below, for a member it will look slightly different),
image
3. As userB, click on the plus actions button in the compose message and select "request money"
4. Once requested have userA and userB click on the expense preview to view the expense report.
5. Confirm userA, the admin sees the subheader, `From <userB full name> in <workspace name>`
image
6. Confirm userB, the member sees the subheader `From <workspace name>`
image
7. Click on the subheader and confirm you are navigated up one level to parent.

### Expense Reports children (Expense Requests)
1. Continuation from the previous section, click on a specific expense request.
2. Confirm both admin and member sees `From <Expense parent report name> in <workspace name>`. The Expense parent report name can be seen in the LHN of the expense report.
3. Create any number of threads or nested threads under the expense report and confirm the subheader for them all is `From <Expense parent report name> in <workspace name>`.
image
4. Click on the subheader and confirm you are always navigated up one level to parent.

For Example, 
251966662-c5fbcaa7-e002-45d1-929a-58b6cc879275

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/38015950/a6d89463-7630-412f-8455-4d0dbcf664d3

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/38015950/81a33598-4b02-4a41-9ac1-0e0e3996cd98

</details>

<details>
<summary>Desktop</summary>

image

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/38015950/bde488f7-6b06-4df8-9642-f31e5471f642

</details>

### CODE REVIEW

**Comment by grgia** [comment]
> I am confused with the number of components that have similiar kind of logic.
> HeaderView.
> HeaderWithbackButton.
> MoneyRequestHeader

I agree with you and thought the same while looking into this. We could refactor here, but I don't think it's a priority. 

> I checked that mainly HeaderView is used for report headers. Why does the change was made to AvatarWithDisplayName?
> Are there only a few types of reports where the header was wrong from the list in the test steps?

AvatarWithDisplayName is only used in the headers, it was using incorrect logic and did not match the LHN.

**Comment by grgia** [comment]
>  Can you please also clarify the changed features? for example, which report-related header is changed.

image

as you can see in this photo, many of the multiple avatars/subscript avatar styles were incorrect and did not match

specifically expense reports, requests, and IOUs are affected

File: src/libs/ReportUtils.js [hunk 1]
```diff
@@ -2422,6 +2422,10 @@ function shouldReportShowSubscript(report) {
         return false;
     }
 
+    if (isMoneyRequestReport(report)) {
```
[the remark this answers is not in the document]
Comment (grgia) [reply src/libs/ReportUtils.js:2425]: Are you looking at a money request or money report while testing? Note, money requests aren't in the LHN unless you have one open currently.

**Comment by grgia** [comment]
Expense Requests and Expense Reports (IOUs) @parasharrajat

**Comment by grgia** [comment]
@parasharrajat here's the example case that this PR fixes - https://github.com/Expensify/App/pull/22562#issuecomment-1638665301

**Comment by JmillsExpensify** [comment]
@parasharrajat next time you log in and focus on /App issues, it would be great to get back to reviewing this issue.

**Comment by grgia** [comment]
@parasharrajat for expense reports, requests, and iou report and requests, I will be adding the the following test steps (will add to description too:



### IOUs
1. Start a DM between userA and userB.
2. Click on the plus actions icon next to "write something.." in the compose message box and select "request money".
3. Click on the expense request preview. 
4. In the expense report, confirm the subheader says `From <userA full name>`
image

### IOUs children  (IOU Requests)
1. Continuing from previous. Click on the specific expense in the expense report.
2. Confirm you see the single expense request view. Confirm the subheader says `From <parent IOU report name>` 
The parent IOU report can be seen in the LHN. In the below screenshot the parent IOU report name is "Marc Glasser owes $20.00" and so the child has that as its subheader.
image
3. Click on the subheader and confirm you are navigated up one level to parent. 
4. Create another thread in the expense report, Or a few nested thread. Confirm all thread subheaders show `From <parent IOU report name>`.
image

### Expense Reports
1. Create a workspace. As admin userA, invite userB to the workspace.
2. You should see a workspace chat between the two like so (view for the admin is below, for a member it will look slightly different),
image
3. As userB, click on the plus actions button in the compose message and select "request money"
4. Once requested have userA and userB click on the expense preview to view the expense report.
5. Confirm userA, the admin sees the subheader, `From <userB full name> in <workspace name>`
image
6. Confirm userB, the member sees the subheader `From <workspace name>`
image
7. Click on the subheader and confirm you are navigated up one level to parent.

### Expense Reports children (Expense Requests)
1. Continuation from the previous section, click on a specific expense request.
2. Confirm both admin and member sees `From <Expense parent report name> in <workspace name>`. The Expense parent report name can be seen in the LHN of the expense report.
3. Create any number of threads or nested threads under the expense report and confirm the subheader for them all is `From <Expense parent report name> in <workspace name>`.
image
4. Click on the subheader and confirm you are always navigated up one level to parent.

**Review by parasharrajat — CHANGES_REQUESTED** [review]

**Comment by grgia** [comment]
@parasharrajat could you explain what you're referring to as wrong in the screenshot

**Comment by grgia** [comment]
@parasharrajat ah yes, the UI you sent is correct. In those screenshots, I am only referring to blue text as "header subtitle"

**Comment by grgia** [comment]
@parasharrajat changes applied

**Comment by grgia** [comment]
@parasharrajat pushed a fix

image

**Comment by grgia** [comment]
@parasharrajat what do you suggest? I can make any updates in this PR

File: src/components/SubscriptAvatar.js [hunk 2]
```diff
@@ -55,6 +55,7 @@ function SubscriptAvatar(props) {
             >
                 <View>
                     <Avatar
+                        containerStyles={StyleUtils.getAvatarStyle(props.size || CONST.AVATAR_SIZE.DEFAULT)}
```
Comment (grgia) [inline src/components/SubscriptAvatar.js:58]: @parasharrajat yeah I agree we should replace this change with the result from this convo  https://github.com/Expensify/App/pull/21657/files#r1275373499

Comment (grgia) [reply src/components/SubscriptAvatar.js:58]: I don't think this change breaks any type of subscript avatar, should we just leave it as is?

Comment (grgia) [reply src/components/SubscriptAvatar.js:58]: @parasharrajat we could just remove the color style here

**Comment by grgia** [comment]
@parasharrajat updated with just height and width

**Comment by grgia** [comment]
@parasharrajat are you able to help get this review to the finish line today?

**Review by parasharrajat — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

:ribbon: :eyes: :ribbon:  C+ reviewed

**Comment by JmillsExpensify** [comment]
@PauloGasparSv Can you treat this PRs as one of your top priorities? It's one of the initiatives that is required before we can close wave3. Thank you!

**Comment by PauloGasparSv** [comment]
> @PauloGasparSv Can you treat this PRs as one of your top priorities? It's one of the initiatives that is required before we can close wave3. Thank you!

Sure @JmillsExpensify, I'm almost done with reviewing+testing https://github.com/Expensify/App/pull/23860 then I'll review this

**Comment by PauloGasparSv** [comment]
Hey, changes look good but I'm getting an error when I try to create a Money Request locally that is blocking me from testing this. I couldn't figure out why this error is happening so **I'll try to finish this review and ask for help if the error persists first thing tomorrow**.

cc @JmillsExpensify in case we need to reassign this to someone else.

image

**Comment by JmillsExpensify** [comment]
Thanks @PauloGasparSv! Let us know how things go tomorrow. If this still isn't resolved for you, then let's get another reviewer in the mix.

**Comment by PauloGasparSv** [comment]
The error is still happening here, I can't create Money Requests. Pretty sure it is unrelated to these changes though.

I think this is happening because the `login` (and therefore `debtorEmail`) is missing in `personalDetailsList`. I'll investigate this a little bit and then **point newDot to staging** if I can't figure out what's happening so we can merge this.

<details>

image

image

<details/>

**Comment by PauloGasparSv** [comment]
Sorry for the delay on this, I'm still stuck on [that error](https://github.com/Expensify/App/pull/22467#issuecomment-1660306488).

I tried pointing newDot to staging but after the `SigninUser` request is successful nothing happens. Now the same is happening when I'm pointing locally.

Btw, I also managed to reproduce the error in staging by requesting money from `002@new.com` specifically (I noticed it didn't have a `login` in `personalDetailsList` either).


@JmillsExpensify @grgia I've assigned puller bear here so another engineer can review while I figure this errors out.

**Comment by PauloGasparSv** [comment]
Had to step out after I assigned another engineer but now I'm back.

Since this is blocking me from doing other stuff I'll give an update here if I fix my local environment and finish the review.

**Comment by JmillsExpensify** [comment]
@PauloGasparSv Can you review the code first? C+ has tested, so I think the review is more important right now.

**Review by PauloGasparSv — APPROVED** [review]
> @PauloGasparSv Can you review the code first? C+ has tested, so I think the review is more important right now.

Sure thing @JmillsExpensify! I already reviewed the changes so I'm approving this based on [these evidences](https://github.com/Expensify/App/pull/22467#issuecomment-1658897786).

Please feel free to merge, still trying to test here.

**Comment by PauloGasparSv** [comment]
I managed to test Web now that I'm aware of https://github.com/Expensify/Expensify/issues/301703 and how to bypass it.

<details>

<summary>Web test</summary>

https://github.com/Expensify/App/assets/6564265/e001a4e2-4779-4dfd-856a-888a803e31f7

</details>

Merging this right now (won't test other platforms to focus on other issues)

**Comment by JmillsExpensify** [comment]
Thanks!

**Pull request merged**